### PR TITLE
Added dtype test concat function

### LINKED ISSUE #32934: Incorrect dtype for concat with empty float and datetime columns
#### Code Sample, a copy-pastable example if possible

```python
In [40]: a = pd.DataFrame({"A": pd.array(['2000'], dtype='datetime64[ns]')})

In [41]: b = pd.DataFrame({"A": pd.array([1.0], dtype='float64')})

In [42]: pd.concat([a.iloc[:0], b.iloc[:0]])['A'].dtype
Out[42]: dtype('<M8[ns]')
```

#### Problem description

The return dtype of concatentating a float64 and datetime64[ns] should be object dtype. That's typically the case when both are non-empty

```python
In [45]: pd.concat([a, b])['A'].dtype
Out[45]: dtype('O')
```

But when either is empty, we get incorrect results

```python
In [42]: pd.concat([a.iloc[:0], b.iloc[:0]])['A'].dtype
Out[42]: dtype('<M8[ns]')

In [43]: pd.concat([a.iloc[:0], b])['A'].dtype
Out[43]: dtype('float64')

In [44]: pd.concat([a, b.iloc[:0]])['A'].dtype
Out[44]: dtype('<M8[ns]')
```

All of those should be object dtype.

### PR DESCRIPTION
**Pull request by GustavoGB**
- [ ] closes #32934 
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/dtypes/test_concat.py [hunk 1]
```diff
@@ -44,3 +44,17 @@ def test_concat_periodarray_2d():
 
     with pytest.raises(ValueError, match=msg):
         _concat.concat_compat([arr[:2], arr[2:]], axis=1)
+
+
+def test_concat_float_datetime():
+
```
Comment (phofl) [inline pandas/tests/dtypes/test_concat.py:50]: Please add gh reference at the top, like GH#issue_number

File: pandas/tests/dtypes/test_concat.py [hunk 2]
```diff
@@ -44,3 +44,17 @@ def test_concat_periodarray_2d():
...
+    df_time = pd.DataFrame({"A": pd.array(["2000"], dtype="datetime64[ns]")})
+    df_float = pd.DataFrame({"A": pd.array([1.0], dtype="float64")})
+
+    assert pd.concat([df_time.iloc[:0], df_float.iloc[:0]])["A"].dtype == object
```
Comment (phofl) [inline pandas/tests/dtypes/test_concat.py:54]: Could you please check the whole DataFrame?

**Review by phofl — COMMENTED** [review]
I think test_datetimes would be a better place for this test?

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in main, address the comments and we can reopen

**Pull request closed without merging**